blueprints/screening_app/__init__.py

```diff
@@ -0,0 +1,420 @@
+    """
+    if 'previous_question_number' not in responder.frame:
+        # Start questionnaire
+        screen_prediabetes(request, responder)
+        return
+
+    previous_question = responder.frame['previous_question_number']
+    question = app.question_answerer.get(index='prediabetes_screening_questions',
+                                         question_number=previous_question)
+
+    question_type = question[0]['type']
+
+    if question_type == 'Binary':
+        set_answer_send_next(request, responder, True)
+        return
+    else:
+        # Answer is out of questionnaire flow. Re-ask question.
+        question_text = responder.frame['previous_question']
+
+        responder.frame['count'] = responder.frame.get('count', 0) + 1
+
+        if responder.frame['count'] <= 3:
+            responder.reply(question_text)
+            responder.listen()
+        else:
+            responder.reply('Disculpe, no le he podido entender. Por favor intente de nuevo.')
+            responder.exit_flow()
+
+
+@screen_prediabetes.handle(intent='answer_no')
+def negate_send_next(request, responder):
+    """
+    When the user answers no to a question, save the answer and move to the next question.
+    """
+    previous_question = responder.frame['previous_question_number']
+    question = app.question_answerer.get(index='prediabetes_screening_questions',
+                                         question_number=previous_question)
+
+    question_type = question[0]['type']
+
+    if question_type == 'Binary':
+        set_answer_send_next(request, responder, False)
+        return
+    else:
+        # Answer is out of questionnaire flow. Re-ask question.
+        question_text = responder.frame['previous_question']
+
+    responder.frame['count'] = responder.frame.get('count', 0) + 1
+
+    if responder.frame['count'] <= 3:
+        responder.reply(question_text)
+        responder.listen()
+    else:
+        responder.reply('Disculpe, no le he podido entender. Por favor intente de nuevo.')
+        responder.exit_flow()
+
+
+@screen_prediabetes.handle(intent='opt_in')
+def screen_prediabetes_in_flow_handler(request, responder):
+    screen_prediabetes(request, responder)
+
+
+@screen_prediabetes.handle(intent='exit')
+def exit_questionnaire(request, responder):
+    """
+    When the user requests to cancel, say goodbye.
+    """
+    say_goodbye(request, responder)
+
+
+@screen_prediabetes.handle(default=True)
+def default_handler(request, responder):
+    responder.frame['count'] = responder.frame.get('count', 0) + 1
+    if responder.frame['count'] <= 3:
+        question_text = responder.frame['previous_question']
+        responder.reply(question_text)
+        responder.listen()
+    else:
+        responder.reply('Disculpe, no le he podido entender. Por favor intente de nuevo.')
+        responder.exit_flow()
+
+
+@app.handle(default=True)
+def default(request, responder):
+    welcome(request, responder)
```

Comment:
You could directly add the `@app.handle(default=True)` decorator to the `welcome` handler, and delete this one. The only difference would be that in this case your response would record the dialogue_state as `default`, so if you were tracking that for metrics it could be helpful. But the user experience would be the same.